*** /scratch/gjb/guile-gtk-0.14/gtk/dynlink.scm	Mon Jan 25 08:17:07 1999
--- /uns/share/guile/gtk/dynlink.scm	Thu Jan 28 10:20:26 1999
***************
*** 318,324 ****
  	      (lib       (if modinfo (caddr modinfo)
  			     (or (dlopen-libtool-library libname)
  				 (error "can't open library" libname)))))
! 	 (if lib
  	     (%sgtk-dlinit init-func lib)
  	     (dynamic-call init-func #f)))))))
  
--- 318,324 ----
  	      (lib       (if modinfo (caddr modinfo)
  			     (or (dlopen-libtool-library libname)
  				 (error "can't open library" libname)))))
! 	 (if (and lib (not modinfo))
  	     (%sgtk-dlinit init-func lib)
  	     (dynamic-call init-func #f)))))))
  
